tests/infer/test_autoguide.py

```diff
@@ -697,6 +697,37 @@ def forward(self, x, y=None):
     assert len(samples) == len(samples_deser)
 
 
+@pytest.mark.parametrize("auto_class", [AutoDelta, AutoNormal])
+def test_subsample_model(auto_class):
+
+    def model(x, y=None, batch_size=None):
+        loc = pyro.param("loc", lambda: torch.tensor(0.))
+        scale = pyro.param("scale", lambda: torch.tensor(1.),
+                           constraint=constraints.positive)
+        with pyro.plate("batch", len(x), subsample_size=batch_size):
+            batch_x = pyro.subsample(x, event_dim=0)
+            batch_y = pyro.subsample(y, event_dim=0) if y is not None else None
+            mean = loc + scale * batch_x
+            sigma = pyro.sample("sigma", dist.LogNormal(0., 1.))
```

Comment:
This is where the test failed prior to @fehiepsi's fix. It seems that local latent variables are needed to exercise the bug.

Reply:
I guess without this local latent variable, we can also test if different runs return different outputs.